biz.aQute.remote/src/aQute/remote/api/Agent.java

```diff
@@ -108,12 +108,13 @@
 	 * differences are reflected in the installed bundles. That is, a change in
 	 * the SHA will update, a new entry will install, and a removed entry will
 	 * uninstall. This is the preferred way to keep the remote framework
-	 * synchronized since it is idempotent.
+	 * synchronized since it is idempotent. Bundles are passed as a
+	 * LinkedHashMap to preserve the startup order.
 	 * 
 	 * @param bundles
 	 *            the bundles to update
 	 */
-	String update(Map<String,String> bundles) throws Exception;
+	String update(LinkedHashMap<String,String> bundles) throws Exception;
```

Comment:
It seems wrong to put an impl specific map type in the API. I understand the implementation will need to use a specific map impl, but the API should not use a specific impl map type.